robotbase/src/main/java/com/team2357/frc2022/shuffleboard/CargoCountWidget.java

```diff
@@ -0,0 +1,65 @@
+package com.team2357.frc2022.shuffleboard;
+
+import com.team2357.frc2022.sensors.SensorBooleanState;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+
+public class CargoCountWidget extends ShuffleboardWidget {
+    private final SensorBooleanState m_feederSensor;
+    private final SensorBooleanState m_intakeSensor;
+    private static NetworkTableEntry m_cargoCountWidget;
+
+    private boolean m_feederStateChange = false;
+    private boolean m_intakeStateChange = false;
+    private int m_numOfCargo;
+
+    public CargoCountWidget(String tabTitle, SensorBooleanState feederSensor, SensorBooleanState intakeSensor) {
+        super(tabTitle);
+        m_feederSensor = feederSensor;
+        m_intakeSensor = intakeSensor;
+
+        m_cargoCountWidget = Shuffleboard.getTab(tabTitle)
+            .add("Num of Cargo Balls", 0)
+            .withWidget(BuiltInWidgets.kTextView)
+            .getEntry();
+    }
+
+    public void addCargo() {
+        int numOfCargo = getNumOfCargo();
+        setNumOfCargo(++numOfCargo);
+        m_cargoCountWidget.setNumber(numOfCargo);
+        m_intakeStateChange = false;
+    }
+
+    public void subCargo() {
+        int numOfCargo = getNumOfCargo();
+        setNumOfCargo(++numOfCargo);
+        m_cargoCountWidget.setNumber(numOfCargo);
+        m_feederStateChange = false;
+    }
+
+
+
+    public void periodic() {
```

Comment:
This periodic function is a bit confusing to me. I think it could be simplified and easier to read. @Tyson453 I'd be happy to pair with you on this.

Reply:
Also, if we're going to go through the trouble of counting our cargo, I'm thinking we may want to use that elsewhere in the robot, so maybe there's a better place for these calculations.